Use SERVER_PORT variable to set listen port

jonseitz opened the pull request:
Previously, we were hard-coding port `8000` into our configs; however, our cloudformation template for launching the app was asking it as a parameter. That is more in line with our usual configuration, so we can update our entrypoint script to use that value instead.

For testing purposes, it should be sufficient to:

1. Add the `SERVER_PORT` to the .env file with a value
2. run `docker-compose up` 
3. Visit localhost:${SERVER_PORT}/admin` and confirm that the login page appears.

jonseitz committed:
Use SERVER_PORT variable to set listen port

Previously, we were hard-coding port 8000 into our configs; however, our
cloudformation template for launching the app was expecting it to be a
parameter. This is more in line with our usual configuration, so we'll
update our entrypoint script to use that value instead.

rmainwork reviewed (changes requested):
Looks good, just a minor thing re environment vars

jonseitz committed:
Remove SERVER_PORT from environment list

Passing the .env file as the env_file argument provides all the
environment variables into the container, so we don't need to include
the SERVER_PORT value under here.

(By contrast, the course-planner project doesn't pass the whole env file
in, so we do need to specify the variables individually there)

jonseitz commented on docker-compose.yml:
```diff
@@ -8,10 +8,11 @@ services:
     env_file: .env
     environment:
       DEBUG: 0
+      SERVER_PORT:
```
Ah, right. Because we've set `env_file: .env` above it's passing all of the environment variables in. We aren't doing that in course-planner, so we did need to specify the individual variables like this in that project, and I think I just carried that over here.

But since it's not needed, I took it out in a0d317d.

rmainseas replied:
Yep. FYI you can also combine the approaches and use `.env` to supply most variables, then use `environment:` to create composite env vars from things in `.env` (if you weren't aware)

rmainwork reviewed (approved):
Looks good. Merge away.
